server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -96,42 +124,68 @@ public void define(Service service) {
             dataType = "String",
             defaultValue = "none",
             example = "user@james.org",
-            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails")
+            value = "Compulsory. Needs to be a valid username represent for an user had requested to restore deleted emails"),
+        @ApiImplicitParam(
+            required = true,
+            dataType = "String",
+            name = "action",
+            paramType = "query",
+            example = "?action=restore",
+            value = "Compulsory. Needs to ve a valid action represent for an operation to perform on the Deleted Message Vault")
     })
     @ApiResponses(value = {
         @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
         @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - user param is invalid"),
         @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side.")
     })
     private TaskIdDto restore(Request request, Response response) throws JsonExtractException {
+        validateActionQuery(request, Action.RESTORE);
+
         User userToRestore = extractUser(request);
         QueryDTO queryDTO = jsonExtractor.parse(request.body());
         Query query = translate(queryDTO);
         TaskId taskId = taskManager.submit(new DeletedMessagesVaultRestoreTask(vaultRestore, userToRestore, query, queryDTO));
         return TaskIdDto.respond(response, taskId);
     }
 
+    private void validateActionQuery(Request request, Action appropriateAction) {
+        try {
+            assertRequestedAction(request, appropriateAction);
+        } catch (IllegalArgumentException e) {
+            haltError(e);
+        }
+    }
+
+    private void assertRequestedAction(Request request, Action appropriateAction) {
+        String actionParam = request.queryParams(ACTION_QUERY_PARAM);
+        Action requestedAction = Action.getAction(actionParam);
+        Preconditions.checkArgument(appropriateAction.equals(requestedAction),
+            "requestedAction " + requestedAction.name().toLowerCase() + " is not allowed(" + appropriateAction.name().toLowerCase() + ")");
```

Comment:
why this `toLowerCase` ?

Reply:
action is case insensitive, and represented by snake uppercase enums. So I think lower case action looks more beautiful than upper snake case regards to the error message send to the client.

Reply:
Who decided that action is case insensitive? Everytime I see such decision, I ask for strict comparison instead so we probably don't have much occurrences of case insensitiveness in the code.

Reply:
I saw some case insensitive actions in the code, so I followed them. But, okay I will update to use case sensitive actions.